core/src/main/java/org/apache/iceberg/ManifestReader.java

```diff
@@ -101,20 +100,33 @@ private String fileClass() {
 
   protected ManifestReader(
       InputFile file,
+      int specId,
       Map<Integer, PartitionSpec> specsById,
       InheritableMetadata inheritableMetadata,
       FileType content) {
     this.file = file;
     this.inheritableMetadata = inheritableMetadata;
     this.content = content;
 
+    if (specsById != null && specsById.containsKey(specId)) {
```

Comment:
Curious, specsById.containsKey(specId) should always be true, shouldnt it?  Does it break if we remove this?